spinalcordtoolbox/types.py

```diff
@@ -394,21 +395,28 @@ def compute_vertebral_distribution(self, discs_levels, label_reference='C1'):
         for i in range(self.number_of_points - 1):
             progress_length[i + 1] = progress_length[i] + self.progressive_length[i]
 
+        # If the reference label (default 'C1' - 1) is not present in the disc labels, then use the uppermost label
+        if label_reference not in self.index_disc.keys():
+            label_reference = self.regions_labels[self.first_label]  # int label -> str label
```

Comment:
This part of the code is just a rewritten, simplified version of the old logic (with `upper = 31`), since I think the old logic was just a convoluted way of determining what the uppermost label is.